sound/soc/intel/boards/Kconfig

```diff
@@ -440,11 +440,12 @@ config SND_SOC_INTEL_GLK_RT5682_MAX98357A_MACH
 
 endif ## SND_SOC_SOF_GEMINILAKE  && SND_SOC_SOF_HDA_LINK
 
+if SND_SOC_INTEL_SKYLAKE_FAMILY || SND_SOC_SOF_HDA
 if SND_SOC_INTEL_SKYLAKE_HDAUDIO_CODEC || SND_SOC_SOF_HDA_AUDIO_CODEC
 
 config SND_SOC_INTEL_SKL_HDA_DSP_GENERIC_MACH
 	tristate "SKL/KBL/BXT/APL with HDA Codecs"
-	select SND_HDA_CODEC_HDMI if SND_SOC_SOF_HDA_AUDIO_CODEC
+	select SND_HDA_CODEC_HDMI if SND_SOC_SOF_HDA && SND_SOC_SOF_HDA_AUDIO_CODEC
```

Comment:
@plbossart  Change looks good, but I don't really grasp how this only affects the generic driver. The kconfig rules look identical to other machine drivers, but somehow the module/builtin conflict only happens here...?

Reply:
the tests where actually different, they only used a boolean for SKL-HDAUDIO, while they used an SOF_GEMINILAKE for example in other places. And what happens is that this removes the possibility of building the machine driver as build-in while SOF is build as a module

It seems the 'if' has side effects, it's not a pure yes/no flow.